### PR TITLE
MAINT Update to Firefox 87 in tests

### PR DESCRIPTION
**Pull request by rth**
Following https://github.com/iodide-project/pyodide/issues/1204#issuecomment-774547684, updates to,
 - Firefox 87 (March 2021)
 - geckodriver 0.29.1 (April 2021)
 - ~~chromedriver 87.0 (Nov 2020) to match the chrome version we are using~~ Done in #1462 

~~Also installs selenium 4.0.0.a7 (Nov 2020).~~ Will be done in https://github.com/pyodide/pyodide/pull/1422

### CODE REVIEW

File: src/pyodide-py/pyodide/webloop.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def run_forever(self):
         """
         pass
 
-    def run_until_complete(self, future: Awaitable):
+    def run_until_complete(self, future: Awaitable):  # type: ignore
```
Comment (rth) [inline src/pyodide-py/pyodide/webloop.py:68]: mypy errors saying that the signature doesn't match the Abstract class, but I don't see any issues there https://github.com/python/cpython/blob/920bf6a3a656e329c2bcbb761eb8c13c46c8cd05/Lib/asyncio/events.py#L214

File: src/pyodide-py/pyodide/webloop.py [hunk 2]
```diff
@@ -223,7 +223,7 @@ def get_task_factory(self):
         return self._task_factory
 
 
-class WebLoopPolicy(asyncio.DefaultEventLoopPolicy):
+class WebLoopPolicy(asyncio.DefaultEventLoopPolicy):  # type: ignore
```
Comment (rth) [inline src/pyodide-py/pyodide/webloop.py:226]: Similarly mypy errors with a message that DefaultEventLoopPolicy is not a valid type.

**Comment by rth** [comment]
In summary, this something that it would be add to do at some point. And it works locally, however it doesn't address the underlying CI issues from #1204. New failures on Firefox would also require more investigation.

**Comment by hoodmane** [comment]
@rth Now that the CI issues are fixed, maybe you should merge master on this one and see if the tests pass?

**Comment by rth** [comment]
Trying to update Firefox only in this PR. We need this if we are to show benchmarks for the 0.17 release, can't really show those with a Firefox released 1.5 years ago (particularly if Chrome is up to date).

**Comment by rth** [comment]
I think I applied workarounds for all failures. So hopefully Firefox CI would pass now, and we would be finally using up to date browsers for tests.

**Comment by hoodmane** [comment]
CI looks good!

**Comment by rth** [comment]
I additionally marked as xfail `test_runwebworker_numpy` for Firefox which wasn't too stable before and seems to be failing more frequently now. We can always investigate later.

**Pull request merged**